source/developers/sts-for-operator.rst

```diff
@@ -0,0 +1,132 @@
+.. _minio-sts-operator:
+
+===============================================
+Security Token Service (STS) for MinIO Operator
+===============================================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+.. versionadded:: Operator v5.0.0
+
+   The MinIO Operator supports a set of API calls that allows an application to obtain STS credentials for a MinIO Tenant.
+
+:ref:`STS credentials <minio-security-token-service>` allow an application to access objects on a MinIO Tenant without the need to create any credentials for the application on the tenant.
+
+.. important:: 
+
+   MinIO Operator launches with STS *disabled* by default.
+   To use STS with the Operator, you must first explicitly enable it.
+
+How STS Authorization Works in Kubernetes
+-----------------------------------------
+
+An application can use an ``AssumeRoleWithWebIdentity`` call to send a request for temporary credentials to the MinIO Operator.
+The Operator checks the validity of the request, retrieves policies for the application, obtains credentials from the tenant, and then passes the credentials back the application.
+The application uses the issued credentials to work with the object storage on the tenant.
+
+The complete process includes the following steps:
+
+1. An application sends an API request to the MinIO Operator containing the name of the application and a service account to use.
+2. The MinIO Operator uses the Kubernetes API to check that the JWT is a valid request.
+3. After confirming the validity of the JWT, the MinIO Operator checks for a ``PolicyBinding`` that matches the application.
+4. The MinIO Operator sends the policy information for the application to the MinIO Tenant.
+5. The tenant creates temporary credentials for the request and returns those to the MinIO Operator.
+6. The MinIO Operator forwards the temporary credentials back to the application.
+7. The application uses the credentials to send the object storage calls to the MinIO tenant.
+
+Requirements
+------------
+
+STS for the MinIO Operator requires the following:
+
+- MinIO Operator v5.0.0 or later
+- Deployment **must** have :ref:`TLS configured <minio-tls>`.
+- :envvar:`OPERATOR_STS_ENABLED` environment variable set to ``on``
+
+Procedure
+---------
+
+1. Enable STS functionality for the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl -n minio-operator set env deployment/minio-operator OPERATOR_STS_ENABLED=on
+   
+   - Replace ``minio-operator`` with the namespace for your deployment.
+   - Replace ``deployment/minio-operator`` with the value for your MinIO deployment
```

Comment:
How would someone know what `the value for your MinIO deployment` should be? Would this be obvious to someone who knows Kubernetes? (Because that person is most assuredly not me.)

Reply:
For the mayority of the deployments, the operator is going to be on the `minio-operator` namespace by default.

On Openshift deployments coming from [OperatorHub](https://operatorhub.io/operator/minio-operator) the default namespace is `openshift-operators`.

Otherwise, users could use a different namespace with the `kubectl minio init --namespace other-namespace` command https://min.io/docs/minio/kubernetes/upstream/reference/kubectl-minio-plugin/kubectl-minio-init.html#kubectl.minio.init.-namespace.

Reply:
Added some instructions here for namespace and deployment values.